storage/src/test/java/tech/pegasys/teku/storage/store/StoreTransactionTest.java

```diff
@@ -404,24 +404,14 @@ public void getOrderedBlockRoots_withNewBlocks() {
     // Added block should be included
     tx.putBlockAndState(forkBlock2, spec.calculateBlockCheckpoints(forkBlock2.getState()));
 
-    // Children are ordered based on hash - so check ordering depending on specific hashes
-    if (mainChainBlock1.getRoot().compareTo(forkBlock2.getRoot()) < 0) {
-      assertThat(tx.getOrderedBlockRoots())
-          .containsExactly(
-              genesis.getRoot(),
-              mainChainBlock1.getRoot(),
-              forkBlock2.getRoot(),
-              mainChainBlock3.getRoot(),
-              mainChainBlock4.getRoot());
-    } else {
-      assertThat(tx.getOrderedBlockRoots())
-          .containsExactly(
-              genesis.getRoot(),
-              forkBlock2.getRoot(),
-              mainChainBlock1.getRoot(),
-              mainChainBlock3.getRoot(),
-              mainChainBlock4.getRoot());
-    }
+    // Childrens are ordered by slot
+    assertThat(tx.getOrderedBlockRoots())
```

Comment:
Not sure about this. the order is not based on slot but on parent->child relationship.
Since `forkBlock2` and `mainChainBlock1` are siblings (both descending from genesis block) they can be picked indifferently (so the root order is applied in this case)

Reply:
I think It is possible to modify the test to hit that condition.

Reply:
I've followed the structure we are queering. The structure is TreeMap with Slot keys, so should be ordered by slots

Reply:
Ah yes, the TreeMap is used when tx data is not empty.